subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/junit/JUnitIntegrationTest.groovy

```diff
@@ -481,7 +481,7 @@ class JUnitIntegrationTest extends JUnitMultiVersionIntegrationSpec {
         if (isVintage() || isJupiter()) {
             result.testClassStartsWith('Gradle Test Executor')
                 .assertTestCount(1, 1, 0)
-                .assertTestFailed("failed to execute tests", containsString("Could not execute test class 'com.example.Foo'"))
+                .assertTestFailed("failed to execute tests", containsString("Could not complete execution for Gradle Test Executor"))
```

Comment:
❓ I think this is not necessary? When I run this locally, the old assertions still does the trick. Or do I miss something?

Reply:
I think you might be missing `-PtestVersions=all`. With just `embeddedIntegTest` it will run this test with JUnit4 only.

Reply:
Why is our setup like this :( it's not different versions in this case....

Anyway, thanks. I now can reproduce it and I see that you choose the generic error so that it is the same for all iterations of the test.

In the error you'll still see a `ClassSelector [className = 'com.example.Foo'] resolution failed`. So the problematic class is still printed. Which is good.